### ISSUE TITLE
`wp_kses_post` in content-widget-product.php breaks lazy-loading plugin

### ISSUE THREAD

**Issue by gerdneuman** [woocommerce/woocommerce#21924]
**Describe the bug**
In https://github.com/woocommerce/woocommerce/commit/4ceee4cfc963166f92d837e357bb1723318a6992 by @rodrigoprimo the html filter on this line was changed from:
```php
 		<?php echo $product->get_image(); ?>
```
to:
```php
 		<?php echo wp_kses_post( $product->get_image() ); ?> 
```
This also filters out the `data-lazy-src` attribute needed by https://wordpress.org/plugins/rocket-lazy-load/
(source at https://github.com/wp-media/rocket-lazy-load/) See also 

I am not sure if there are other `data-*` attributes needed by the lazy-load plugin, e.g. `data-no-lazy` or more. Search the following file for `data-` more references:
https://github.com/wp-media/rocket-lazy-load/blob/3f196b9b4a2e52b750b222a740876d88e360b79d/rocket-lazy-load.php

CC @Tabrisrp from WP-Rocket

**To Reproduce**
1. Install https://wordpress.org/plugins/rocket-lazy-load/
2. Add `Products` Widget in the sidebar (https://docs.woocommerce.com/document/woocommerce-widgets/#section-3)
3. `<img>` is output with empty `src` (expected) but without `data-lazy-src` (unexpected => regression bug)
4. No images are loaded in the Products widget 

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Proposed solution:**
Easiest (and I don't see security issues) would be to allow all `data-*` attributes when filtering with `wp_kses_post` in 
https://github.com/woocommerce/woocommerce/blob/d3c67b675b8f8364a06367105cf3631b8135392c/templates/content-widget-product.php#L32 (possibly also at other occurences). Not sure though how this would be done (seems like `wp_kses_post` does not accept any params to whitelist certain attributes…)

**Comment by kloon** [comment]
Issue introduced in PR #20881

**Comment by kloon** [comment]
Fix in #21929
